flask_sillywalk/sillywalk.py

```diff
@@ -111,28 +108,121 @@ def inner_func(c, *args, **kwargs):
                 if self.models[c.__name__].get("required") is None:
                     self.models[c.__name__]["required"] = []
                 self.models[c.__name__]["required"].append(arg)
-                #self.models[c.__name__]["required"][arg] = {"required": True}
+                # self.models[c.__name__]["required"][arg] = {"required": True}
             for k, v in defaults:
                 self.models[c.__name__]["properties"][k] = {"default": v}
             return c
 
         return inner_func
 
-    def register(
-            self,
+    def _register(self,
+                  path,
+                  f,
+                  method,
+                  content_type,
+                  parameters,
+                  responseMessages,
+                  nickname,
+                  notes,
+                  bp):
+
+        if self.app is None:
+            raise SwaggerRegistryError(
+                "You need to initialize {0} with a Flask app".format(
+                    self.__class__.__name__))
+
+        # use basepath if not set by user
+        if self.basepath not in path:
+            path = self.basepath + "/" + path
+            path = path.replace("//", "/")
+
+        # register views on blueprints
+        app = self.app
+
+        if bp:
+            app = bp
+
+        app.add_url_rule(
             path,
-            method="GET",
-            content_type="application/json",
-            parameters=[],
-            responseMessages=[],
-            nickname=None,
-            notes=None):
+            f.__name__,
+            f,
+            methods=[method])
+
+        api = Api(
+            method=f,
+            path=path.replace(self.basepath, ""),
+            httpMethod=method,
+            params=parameters,
+            responseMessages=responseMessages,
+            nickname=nickname,
+            notes=notes)
+
+        if api.resource not in self.app.view_functions:
+            for fmt in SUPPORTED_FORMATS:
+                route = "{0}/{1}.{2}".format(self.basepath.rstrip("/"),
+                                             api.resource, fmt)
+                if route not in self.registered_routes:
+                    self.registered_routes.append(route)
+                    self.app.add_url_rule(
+                        route,
+                        api.resource,
+                        self.show_resource(api.resource))
+
+        if self.r[api.resource].get(api.path) is None:
+            self.r[api.resource][api.path] = list()
+        self.r[api.resource][api.path].append(api)
+
+    def add_register(self,
+                     path,
+                     f,
+                     method="GET",
+                     content_type="application/json",
+                     parameters=[],
+                     responseMessages=[],
+                     nickname=None,
+                     notes=None,
+                     bp=None):
+        """
+        Registers an API endpoint.
```

Comment:
To support dynamic adding of urls. Its the same like ``app.add_url_rule('/', 'index', index)`` and ``@app.route('/')``